src/Components/ImagesTable/ImagesTable.tsx

```diff
@@ -71,28 +78,56 @@ const ImagesTable = () => {
     setPerPage(perPage);
   };
 
-  const { data, isSuccess } = useGetComposesQuery({
-    limit: perPage,
-    offset: perPage * (page - 1),
-    ignoreImageTypes: [
-      'rhel-edge-commit',
-      'rhel-edge-installer',
-      'edge-commit',
-      'edge-installer',
-    ],
-  });
+  const {
+    data: blueprintsComposes,
+    isSuccess: isBlueprintsSuccess,
+    isFetching: isFetchingBlueprintsCompose,
+  } = useGetBlueprintComposesQuery(
+    {
+      id: selectedBlueprint || '',
```

Comment:
I assume we need `''` here because `selectedBlueprint` could be undefined and the linter is unhappy about that, because it doesn't realize that the `skip` will prevent it from happening. I've used similar patterns before but I'm not quite happy with them. What if we tried using a lazy query instead?

https://redux-toolkit.js.org/rtk-query/api/created-api/hooks#uselazyquery

We could call the lazy query's trigger function only if a selectedBlueprint was present.

Thoughts?

Reply:
Agree, I'm not happy with it as well.
The `useLazyQuery` looks interesting, but  it creates a trigger function, so the query won't start once the component renders, so we can trigger it inside an event handler . In ours case, we want to fetch the data once the table renders, so I'd pick `skip` here.  because we use `selectedBlueprint` in the skip, we can be certain that it string, Se can do type assertion:

```js
useGetBlueprintComposesQuery(
    {

      id: selectedBlueprint as string, // thanks to skip - `selectedBlueprint` must be string 
      limit: perPage,
      offset: perPage * (page - 1),
    },
    { skip: !selectedBlueprint }
  );
```

Reply:
I think that's a lot cleaner and definitely prefer it. Let's go with that for now, I'll think about the implications of the trigger function a bit more on my own.